base/array.jl

```diff
@@ -252,9 +252,14 @@ function bitsunionsize(u::Union)
     return sz
 end
 
-# Deprecate this, as it seems to have no documented meaning and is unused here,
-# but is frequently accessed in packages
 elsize(@nospecialize _::Type{A}) where {T,A<:Array{T}} = aligned_sizeof(T)
+function elsize(::Type{Ptr{T}}) where T
+    # this only must return something valid for values which satisfy is_valid_intrinsic_elptr(T),
+    # which includes Any and most concrete datatypes
+    T === Any && return sizeof(Ptr{Any})
+    T isa DataType || sizeof(Any) # throws
+    return LLT_ALIGN(Core.sizeof(T), datatype_alignment(T))
+end
```

Comment:
Can you add a couple tests for `elsize(Ptr{Any})`, `elsize(Ptr{Ref{Any}})`, `elsize(Ptr{NTuple{3,Int8}})`, `elsize(Ptr)`, `elsize(Ptr{Int})`, and `elsize(Ptr{Cvoid})`